api/src/main/java/org/openmrs/validator/VisitValidator.java

```diff
@@ -159,6 +165,44 @@ private void validateStopDatetime(Visit visit, Visit otherVisit, Errors errors)
 			errors.rejectValue("stopDatetime", "Visit.visitCannotContainAnotherVisitOfTheSamePatient", messageBuilder
 			        .toString());
 		}
-		
+
+	}
+
+	private void validateStartDateTimeToBirthdate(Visit visit, Errors errors) {
+
+		Patient patient = visit.getPatient();
+
+		if (patient != null) {
+			if (visit.getStartDatetime() != null && patient.getBirthdate() != null) {
```

Comment:
Same here, startDatetime is required and was checked earlier, I understand why he checks why it's null here but it's a result of this validator not being well written, in theory a validator needs to check required field first at the top like this does and then only proceeds if they are all set, kind of like the snippet below:
```
ValidationUtils.rejectIfEmpty(errors, "patient", "Visit.error.patient.required");
ValidationUtils.rejectIfEmpty(errors, "visitType", "Visit.error.visitType.required");
ValidationUtils.rejectIfEmpty(errors, "startDatetime", "Visit.error.startDate.required");
if(errors.hasErrors()){
    return;
}
//At this point we don't need any null checks for the required fields
```